src/meta/src/rpc/service/ddl_service.rs

```diff
@@ -630,15 +636,25 @@ where
                     .finish_create_sink_procedure(sink)
                     .await?
             }
-            StreamingJob::MaterializedSource(source, table) => {
+            StreamingJob::Table(source, table) => {
                 creating_internal_table_ids.push(table.id);
-                self.catalog_manager
-                    .finish_create_materialized_source_procedure(
-                        source,
-                        table,
-                        &ctx.internal_tables()[0],
-                    )
-                    .await?
+                if let Some(source) = source {
+                    let internal_tables = ctx.internal_tables();
+                    assert_eq!(internal_tables.len(), 1);
+                    self.catalog_manager
+                        .finish_create_table_procedure_with_source(
+                            source,
+                            table,
+                            &internal_tables[0],
+                        )
+                        .await?
+                } else {
+                    let internal_tables = ctx.internal_tables();
+                    assert!(internal_tables.is_empty());
+                    self.catalog_manager
+                        .finish_create_table_procedure(ctx.internal_tables(), table)
```

Comment:
They are almost the same, except that `create table` doesn't have any internal table. I think it's okay to reuse this method, since `create table` will simply skip iterating over `internal_tables` in this method.

Reply:
It's okay for me to reuse it, would be better to add some comment.